packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,232 @@
+        setTarget(_owner);
+        transferOwnership(_owner);
+
+        emit OptimisticOracleModuleDeployed(_owner, avatar, target);
+    }
+
+    function setBond(uint256 _bond) public onlyOwner {
+        // value of the bond required for proposals, in addition to the final fee
+        bond = _bond;
+    }
+
+    function setCollateral(IERC20 _collateral) public onlyOwner nonReentrant() {
+        // ERC20 token to be used as collateral (must be approved by UMA Store contract).
+        require(_getCollateralWhitelist().isOnWhitelist(address(_collateral)), "bond token not supported");
+        collateral = _collateral;
+    }
+
+    function setRules(string memory _rules) public onlyOwner {
+        // set reference to the rules for the avatar (e.g. an IPFS hash or URI)
+        rules = _rules;
+    }
+
+    function setLiveness(uint64 _liveness) public onlyOwner {
+        // set liveness for disputing proposed transactions
+        require(_liveness > 0, "liveness can't be 0");
+        liveness = _liveness;
+    }
+
+    function sync() public nonReentrant {
+        _sync();
+    }
+
+    function proposeTransactions(Transaction[] memory _transactions, bytes memory _explanation) public nonReentrant() {
+        // create a proposal with a bundle of transactions
+        // note: based in part on the UMA Governor contract
+        // https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/implementation/Governor.sol
+        // note: optional explanation explains the intent of the transactions to make comprehension easier
+        uint256 id = proposals.length;
+        uint256 time = block.timestamp;
+        address proposer = msg.sender;
+
+        // Add a zero-initialized element to the proposals array.
+        Proposal storage proposal = proposals.push();
+        proposal.requestTime = time;
+        proposal.ancillaryData = _explanation;
+
+        // Construct the ancillary data.
+        AncillaryData.appendKeyValueUint(proposal.ancillaryData, "id", id);
+        AncillaryData.appendKeyValueAddress(proposal.ancillaryData, "module", address(this));
+
+        // Initialize the transaction array.
+        for (uint256 i = 0; i < _transactions.length; i++) {
+            require(_transactions[i].to != address(0), "The `to` address cannot be 0x0");
+            // If the transaction has any data with it the recipient must be a contract, not an EOA.
+            if (_transactions[i].data.length > 0) {
+                require(_isContract(_transactions[i].to), "EOA can't accept tx with data");
+            }
+            proposal.transactions.push(_transactions[i]);
+        }
+
+        // Propose a set of transactions to the OO. If not disputed, they can be executed with executeProposal().
+        // docs: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/interfaces/SkinnyOptimisticOracleInterface.sol
+        _getOptimisticOracle().requestAndProposePriceFor(
```

Comment:
where is the bonding logic? do you not need to pull the `bondAmount` from the caller?

Reply:
I think so, will add that and make sure it's tested before requesting a re-review.